### DISCUSSION TITLE
Aligning 2 meshes using ICP

### DISCUSSION THREAD

**Discussion by cyubit** [pyvista/pyvista#4310] [Q&A] [answered]
I have 2 different but similar meshes that i would like to align as closely as possible. Someone recommended using an iterative closest point method but I can't find anything about this in the pyvista documentation. Any suggestions? 

<img width="392" alt="Screen Shot 2023-04-20 at 1 43 11 PM" src="https://user-images.githubusercontent.com/87676972/233446049-ebd06bf9-b7e9-45ac-9309-c78e7dfb8092.png">

**Comment by darikg** [comment, marked as answer]
There's an example of using [vtkIterativeClosestPointTransform](https://vtk.org/doc/nightly/html/classvtkIterativeClosestPointTransform.html) in python [here](https://kitware.github.io/vtk-examples/site/Python/PolyData/AlignTwoPolyDatas/)

Looks like the bare minimum is something like

```python
from vtkmodules.vtkCommonDataModel import vtkIterativeClosestPointTransform

icp = vtkIterativeClosestPointTransform()
icp.SetSource(mesh_a)
icp.SetTarget(mesh_b)
icp.GetLandmarkTransform().SetModeToRigidBody()
icp.SetMaximumNumberOfLandmarks(100)
icp.SetMaximumMeanDistance(.00001)
icp.SetMaximumNumberOfIterations(500)
icp.CheckMeanDistanceOn()
icp.StartByMatchingCentroidsOn()
icp.Update()

aligned = mesh_a.transform(icp.GetMatrix())
```

**Reply by cyubit** [reply]
This worked great I got the result I wanted, thank you!
